packages/matchbox/src/components/Button/Button.js

```diff
@@ -1,140 +1,147 @@
+
+    // Events
+    onClick,
+    onFocus,
+    onBlur,
+
+    className = '',
+    ...rest // TODO remove spreading of unknown props
+  } = props;
+
+  // Polyfills deprecrated 'Component' prop
+  const WrapperComponent = component || Component;
+
+  // Polyfills to be deprecrated 'primary' and 'destructive' prop
+  const buttonColor = primary ? 'blue' : destructive ? 'red' : color;
+
+  // Experimenting with a weight prop to replace outline, plain, and flat in the future
+  const visualWeight = React.useMemo(() => {
+    if (outline) {
+      return 'normal';
+    }
 
-  handleMouseUp({ currentTarget }) {
-    currentTarget.blur();
-  }
+    if (plain || flat) {
+      return 'weak';
+    }
 
-  render() {
-    const {
-      children,
-
-      // Styles
-      primary, // Deprecate in favor of color
-      color,
-      disabled,
-      destructive,
-      plain, // Deprecate in favor of flat
-      flat,
-      outline,
-
-      // Options
-      size,
-      fullWidth,
-      submit,
-
-      to,
-      Component,
-      component,
-      external,
-
-      // Events
-      onClick,
-      onFocus,
-      onBlur,
-
-      className = '',
-      ...rest
-    } = this.props;
-
-    const WrapperComponent = component || Component;
-
-    const buttonColor = primary ? 'orange' : color;
-
-    const classname = classnames(
-      styles.Button,
-      buttonColor && styles[`color-${buttonColor}`],
-      disabled && styles.disabled,
-      destructive && styles.destructive,
-      (flat || plain) && styles.flat,
-      outline && styles.outline,
-      fullWidth && styles.fullWidth,
-      (size && size !== 'default') && styles[`${size}`],
-      className
-    );
+    return 'strong';
+  }, [outline, plain, flat]);
 
-    if (to && !WrapperComponent) {
-      return (
-        <a
-          href={to}
-          target={external ? '_blank' : ''}
-          rel={external ? 'noopener noreferrer' : ''}
-          onClick={onClick}
-          onFocus={onFocus}
-          onBlur={onBlur}
-          className={classname}
-          disabled={disabled}
-          onMouseUp={this.handleMouseUp}
-          {...rest}
-        >
-          {children}
-        </a>
-      );
-    }
+  const sharedProps = {
+    className, disabled, fullWidth, onClick, onFocus, onBlur, buttonSize, visualWeight, buttonColor, ...rest
+  };
 
-    if (to && WrapperComponent) {
-      return (
-        <WrapperComponent
-          to={to}
-          onClick={onClick}
-          onFocus={onFocus}
-          onBlur={onBlur}
-          className={classname}
-          disabled={disabled}
-          onMouseUp={this.handleMouseUp}
-          {...rest}
-        >
-          {children}
-        </WrapperComponent>
-      );
-    }
+  if (to && !WrapperComponent) {
+    return (
+      <StyledButton
+        as='a'
+        href={to}
+        target={external ? '_blank' : ''}
+        rel={external ? 'noopener noreferrer' : ''}
+        title={external ? 'Opens in a new tab' : ''}
```

Comment:
It looks like this renders an empty `title` attribute when the link is not `external`.

Maybe we could have a `title` prop that would typically be undefined. That would handle the aforementioned empty title problem *and* allow some flexibility:

```
const { title } = props;

<Button title={external ? 'Opens in a new tab' : title}/>
```